api/v2/routes_rtfs.go

```diff
@@ -381,8 +375,11 @@ func (api *API) extendPin(c *gin.Context) {
 		api.LogError(c, err, eh.UploadSearchError)(http.StatusBadRequest)
 		return
 	}
+	// indicate whether or not the user is free
+	// will be true if usg is free tier
+	var isFree = usg.Tier == models.Free
 	// ensure even with pin time extension, it wont breach two year limit
-	if err := api.ensureTwoYearMax(upload, holdTimeInt); err != nil {
+	if err := api.ensureLEMaxPinTime(upload, holdTimeInt, isFree); err != nil {
```

Comment:
If the function signature is changed to accept `usg.Tier` directly, it's better for type safety and future-proofing possible changes to max pin time.

Reply:
smart